index/scorch/scorch.go

```diff
@@ -641,6 +656,31 @@ func (s *Scorch) StatsMap() map[string]interface{} {
 	m["num_persister_nap_merger_break"] = m["TotPersisterMergerNapBreak"]
 	m["total_compaction_written_bytes"] = m["TotFileMergeWrittenBytes"]
 
+	fieldStats := make(map[string]map[string]int)
+
+	for _, segmentSnapshot := range indexSnapshot.Segments() {
+		if segmentSnapshot.stats != nil {
+			for statName, stats := range segmentSnapshot.stats {
+				if _, exists := fieldStats[statName]; !exists {
+					fieldStats[statName] = make(map[string]int)
+				}
+
+				for fieldName, val := range stats {
+					if _, exists := fieldStats[statName][fieldName]; !exists {
+						fieldStats[statName][fieldName] = 0
+					}
+
+					fieldStats[statName][fieldName] += val
+				}
+			}
+		}
+	}
+
+	for statName, stats := range fieldStats {
+		for fieldName, val := range stats {
+			m["field:"+fieldName+":"+statName] = val
```

Comment:
I understand that this is a per-field stat but I'm not sure adding the "field" prefix here is the best idea?

Reply:
I have discussed this with Abhi and we thought this was a good idea. Open to suggestions though